recipes/vulkan-loader/all/conanfile.py

```diff
@@ -0,0 +1,151 @@
+import os
+import glob
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+
+
+class VulkanLoaderConan(ConanFile):
+    name = "vulkan-loader"
+    description = "Khronos official Vulkan ICD desktop loader for Windows, Linux, and MacOS."
+    topics = ("conan", "vulkan", "loader", "desktop", "gpu")
+    homepage = "https://github.com/KhronosGroup/Vulkan-Loader"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_wsi_xcb": [True, False],
+        "with_wsi_xlib": [True, False],
+        "with_wsi_wayland": [True, False],
+        "with_wsi_directfb": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_wsi_xcb": True,
+        "with_wsi_xlib": True,
+        "with_wsi_wayland": True,
+        "with_wsi_directfb": False,
+    }
+
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "pkg_config"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os != "Linux":
+            del self.options.with_wsi_xcb
+            del self.options.with_wsi_xlib
+            del self.options.with_wsi_wayland
+            del self.options.with_wsi_directfb
+        # default shared on non-Apple OS
+        if not tools.is_apple_os(self.settings.os):
+            self.options.shared = True
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.get_safe("with_wsi_directfb"):
+            # TODO: directfb package
+            raise ConanInvalidConfiguration("Conan recipe for DirectFB is not available yet.")
+        if not tools.is_apple_os(self.settings.os) and not self.options.shared:
+            raise ConanInvalidConfiguration("Static builds are not supported on {}".format(self.settings.os))
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
+            # FIXME: It should build but Visual Studio 2015 container in CI of CCI seems to lack some Win SDK headers
+            raise ConanInvalidConfiguration("Visual Studio < 2017 not yet supported in this recipe")
+
+    def requirements(self):
+        self.requires("vulkan-headers/{}".format(self.version))
```

Comment:
Does validate support deps versioning?

Reply:
A: https://github.com/conan-io/conan-center-index/pull/4308/files#r560274484

Reply:
And no, `validate()` doesn't support deps versions yet 😉

Reply:
What about `self.deps_cpp_info[“vulkan-headers”].version`? TBH, I haven't tried it, and the interface to get information from other nodes in the graph will change in Conan 2.0 (we want something more systematic), but for the moment it might be a good alternative.